src/components/ui/card/samInfoCard.tsx

```diff
@@ -38,12 +38,31 @@ export const SamInfoCard = ({
   const [maxValue, setMaxValue] = useState<number | string>('N/A');
   const [hov, setHov] = useState<boolean>(false);
 
+  const formatNumber = (input: number | string) => {
+    const num = Number(input);
+    if (isNaN(num)) return input.toString();
+
+    if (Math.abs(num) >= 1e5) {
+      return num.toExponential(2);
+    }
+    return Number(num.toFixed(3)).toString();
+  };
+
+  const extractDateTime = (input: string): string => {
+    const match = input.match(/\d{4}-\d{2}-\d{2}T\d{6}Z/);
```

Comment:
use naming convention of stac item id to get datetime i.e. by using process of elimination to get datetime.